### ISSUE TITLE
Edition des cellules en double cliquant dessus

### ISSUE THREAD

**Issue by julienCsj** [FACT-Team/FactDev#215]
Il serait très agréable de pouvoir éditer les cellules en cliquant directement dessus.

**Comment by aroquemaurel** [comment]
Cela me semble raisonnable.

**Comment by aroquemaurel** [comment]
Pour pouvoir faire cette issue, il faut revoir l'ux… @Oxynos @manantsoa @KraTuX31 

Si on fait un clique clique, ben ça fait passer de clients à Projet, puis clique clique sur un projet vers Factures.
Or, M. @julienCsj, souhaiterai que si on fait un clique clique sur n'importe quoi, ça permette l'édition. 
Dans ce contexte là, ça ne peut marcher :-(

**Comment by julienCsj** [comment]
Dois-je comprendre que la feature ne sera pas implementé ?

**Comment by aroquemaurel** [comment]
Il faut y réfléchir surtout ^^
Mais je dois dire que du coup, c'est un peu plus lourd que prévu, et donc, ayant un super Projet d'AL qui m'intéresse bien plus… :Ð
Après, si on a une solution pour passer de client en projet vers facture, facilement, compréhensible autre que le double clique, je dis pas non ;)

**Comment by julienCsj** [comment]
Eventuellement un "Clic droit > Editer la cellule", mais c'est moins naturel je trouve.
Du coup il faut aussi changer le label "Easy" ;)

**Comment by aroquemaurel** [comment]
Yep, j'y ai pensé… Actuellement un clique droit « editer » ouvre une fenêtre d'édition quoi ^^
Du coup, je sais pas trop.
J'vais y réfléchir :p 

Exact ! :p

**Comment by Oxynos** [comment]
Mais en gros c'est exactement comme dans le logiciel sqllitebrowser fait avec QT, on aurait pu réutiliser le code sur github, enfin c'est beaucoup de travail

**Comment by aroquemaurel** [comment]
Gné ?

Et en soit, l'édition au clique clique, c'est pas grand chose (déjà fait dans les prestations), c'est juste en therme d'ux ou ça chie.

**Comment by aroquemaurel** [comment]
Bon, désolé mon ptiti @julienCsj, mais ça sera pas fait tout de suite… Je garde ça sous le coude parce que ça serait cool, mais ce n'est pas dans les trucs les plus prioritaires… ;-)

**Comment by julienCsj** [comment]
Déception :( 
Je vais faire une PR dans ce cas !

**Comment by aroquemaurel** [comment]
M. @superkebab a trouvé une solution satisfaisante, votre issue sera donc implémentée :-)
L'idée étant de passer par un `selectedclick` et non par un `doubleclicked` :)

**Comment by julienCsj** [comment]
Hourra \o/

**Comment by superkebab** [comment]
Je suis vraiment très très fort, non ?

**Comment by aroquemaurel** [comment]
![](http://m.memegen.com/8pctqw.jpg)

**Comment by superkebab** [comment]
J'aime

**Comment by aroquemaurel** [comment]
Précisions sur le comment on peut implémenter ça (pour @KraTuX31): 

> **(17:01:51) Antoine:** Il faut que pour chaque tableview tu mette un edit trigger à selectedclick (tu peux changer ça depuis le designer).
> Ensuite, normalement le setData de chacun des modèles sera appelé : vérifier qu'ils sont bons, et si tel est le cas, enchainer avec un commit();
> Après, faut voir aussi parce qu'il y a des cellules qui ne sont pas autorisées à l'édition (Numéro de Facture, Dates, payée, …), pour ces cellules, ben il faut empêcher cette édition ^^
> **(17:02:55) Antoine:** Du coup, avec le selectedClick, tu pourras éditer ta cellule quand, tu sélectionne une ligne, puis tu appuie sur ta cellule, ça ouvrira l'édition, le setData sera appelé quand tu désélectionnera ta cellule (entrée, clique à côtés, etc…)
